glean-core/src/lib.rs

```diff
@@ -394,6 +397,15 @@ fn initialize_inner(
             log::set_max_level(level)
         }
 
+        let data_path_str = cfg.data_path.clone();
+        let data_path = Path::new(&data_path_str);
+        let internal_pings_enabled = cfg.enable_internal_pings;
+        let dir_info = if !is_test_mode() && internal_pings_enabled {
+            collect_directory_info(Path::new(&data_path))
```

Comment:
Bug: This uses `core::with_glean` for the error metrics, but `core::with_glean` will not work (will panic `Global Glean object not initialized`) until `core::setup_glean`, below.

Reply:
Oddly, I never saw this panic with any of the tests or examples